lighthouse-core/gather/driver.js

```diff
@@ -86,6 +86,14 @@ class Driver {
       this._handleReceivedMessageFromTarget(event, []).catch(this._handleEventError);
     });
 
+    this.on('Runtime.executionContextDestroyed', event => {
+      if (event.executionContextId === this._isolatedExecutionContextId) {
+        this._clearIsolatedContextId();
+      }
+    });
+
+    this.on('Page.frameNavigated', this._clearIsolatedContextId.bind(this));
```

Comment:
```suggestion
    this.on('Page.frameNavigated', () => this._clearIsolatedContextId);
```
?

Reply:
oops that's not right though haha, but point taken :)